acai-ios/ViewControllers/LoginViewController.swift

```diff
@@ -122,71 +108,184 @@ class LoginViewController: UIViewController {
         ]
     }
 
-    func switchDataEntry(to type: EntryType) {
-        if type == .login {
-            emailInputView.snp.remakeConstraints { make in
-                make.leading.equalToSuperview().offset(FileConstants.leadingOffset)
-                make.trailing.equalToSuperview().offset(-FileConstants.trailingOffset)
-                make.height.equalTo(FileConstants.inputViewHeight)
-                make.top.equalTo(view.safeAreaLayoutGuide.snp.top).offset(FileConstants.inputViewTopOffset)
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return inputItems.count + 1
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = indexPath.row == inputItems.count ? tableView.dequeueReusableCell(withIdentifier: FileConstants.roundedButtonCellReuseIdentifier, for: indexPath) as! RoundedButtonTableViewCell : tableView.dequeueReusableCell(withIdentifier: FileConstants.inputCellReuseIdentifier, for: indexPath) as! InputTableViewCell
+        if let cell = cell as? InputTableViewCell {
```

Comment:
instead of doing this check (`cell = cell as? InputTableViewCell`) I think its better to just do
```
if indexPath.row == inputItems.count {
   let cell = ...
    ...
    return cell
}
let cell = ...
...
return cell
```